### PR TITLE
test objects deletion from a OBC when there is huge number of objects ~ 2M

### PR DESCRIPTION
**Pull request by mashetty330**
bug: https://bugzilla.redhat.com/show_bug.cgi?id=2181535

### CODE REVIEW

File: ocs_ci/ocs/bucket_utils.py [hunk 1]
```diff
@@ -676,10 +679,29 @@ def cli_create_ibmcos_backingstore(
...
+        cld_mgr (CloudManager): holds secret for backingstore creation
+        backingstore_name (str): backingstore name
+        uls_name (str): underlying storage name
+        region (str): which region to create backingstore (should be the same as uls)
+    """
     pass
 
 
-def cli_create_s3comp_backingstore(cld_mgr, backingstore_name, uls_name, region):
+def cli_create_s3comp_backingstore(
```
Comment (parikshithb) [inline ocs_ci/ocs/bucket_utils.py:693]: @mashetty330 is this change necessary?

File: tests/e2e/scale/noobaa/test_delete_objects.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+class TestDeleteObjects:
+    @bugzilla("2181535")
+    @polarion_id("OCS-4916")
+    @pytest.mark.parametrize(
+        argnames=["delete_mode"],
+        argvalues=[
+            pytest.param("single"),
+            # pytest.param("batch"),
+            # pytest.param("whole"),
```
Comment (parikshithb) [inline tests/e2e/scale/noobaa/test_delete_objects.py:37]: need to uncomment these?

File: tests/e2e/scale/noobaa/test_delete_objects.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+        bucket_class_dict = {
+            "interface": "OC",
+            "backingstore_dict": {"aws": [(1, "eu-central-1")]},
+        }
+        # create an object bucket
+        bucket = bucket_factory(bucketclass=bucket_class_dict, verify_health=False)[0]
+        bucket.verify_health(timeout=300)
+
+        # write 4K and 24M size objects of 1M each to the bucket
```
Comment (parikshithb) [inline tests/e2e/scale/noobaa/test_delete_objects.py:63]: this needs to be 4K and 4M instead of 24M?

**Review by parikshithb — COMMENTED** [review]
few comments, overall looks great!

File: ocs_ci/ocs/bucket_utils.py (hunk 1 above)
Comment (mashetty330) [reply ocs_ci/ocs/bucket_utils.py:702]: yeah please ignore that

File: tests/e2e/scale/noobaa/test_delete_objects.py (hunk 2 above)
Comment (mashetty330) [reply tests/e2e/scale/noobaa/test_delete_objects.py]: yes

File: tests/e2e/scale/noobaa/test_delete_objects.py (hunk 3 above)
Comment (mashetty330) [reply tests/e2e/scale/noobaa/test_delete_objects.py]: correct

**Comment by mashetty330** [comment]
This bug https://bugzilla.redhat.com/show_bug.cgi?id=2225008 is blocking the validation. Cluster goes to bad state after the first iteration of test is run making the rest of the iteration fail.

File: tests/e2e/scale/noobaa/test_delete_objects.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+@scale
+class TestDeleteObjects:
+    @bugzilla("2181535")
+    @polarion_id("OCS-4916")
+    @pytest.mark.parametrize(
+        argnames=["delete_mode"],
+        argvalues=[
+            # uncomment the blow params when we have stable noobaa db
+            # or work around for memory issue caused during single deletion
+            # pytest.param("single"),
```
Comment (PrasadDesala) [inline tests/e2e/scale/noobaa/test_delete_objects.py:39]: It would be good to also add the BZ/issue tracking that issue here as a reference and to take an action to uncomment the code once the BZ is fixed

Comment (mashetty330) [reply tests/e2e/scale/noobaa/test_delete_objects.py]: @PrasadDesala I have added a comment mentioning what is the issue. There is no bz or issue pertaining to this.

File: tests/conftest.py [hunk 5]
```diff
@@ -7296,3 +7306,133 @@ def finalizer():
...
+        '{"spec": {"multiCloudGateway": {"endpoints": {"minCount": 3,"maxCount": 10}}}}'
+    )
+    scale_noobaa_resources_param = (
+        '{"spec": {"resources": {"noobaa-core": {"limits": {"cpu": "3","memory": "4Gi"},'
+        '"requests": {"cpu": "3","memory": "4Gi"}},"noobaa-db": {"limits": {"cpu": "3","memory": "4Gi"},'
+        '"requests": {"cpu": "3","memory": "4Gi"}},"noobaa-endpoint": {"limits": {"cpu": "3","memory": "4Gi"},'
+        '"requests": {"cpu": "3","memory": "4Gi"}}}}}'
+    )
+    storagecluster_obj.patch(params=scale_endpoint_pods_param, format_type="merge")
+    storagecluster_obj.patch(params=scale_noobaa_resources_param, format_type="merge")
```
[the remark this answers is not in the document]
Comment (mashetty330) [reply tests/conftest.py:7341]: ack

File: ocs_ci/ocs/resources/backingstore.py [hunk 6]
```diff
@@ -280,7 +280,7 @@ def backingstore_factory(request, cld_mgr, mcg_obj, cloud_uls_factory):
         },
     }
 
-    def _create_backingstore(method, uls_dict):
+    def _create_backingstore(method, uls_dict, timeout=None):
```
Comment (fbalak) [inline ocs_ci/ocs/resources/backingstore.py:283]: New parameter is missing a docstring.

File: tests/cross_functional/scale/noobaa/test_delete_objects.py [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+    @bugzilla("2181535")
+    @polarion_id("OCS-4916")
+    @pytest.mark.parametrize(
+        argnames=["delete_mode"],
+        argvalues=[
+            # Below parameter is commented now because single deletion is causing
+            # heavy toll on memory consumption for the jenkins agent and test run time,
+            # as we have to delete one object at a time, till 2 million objects are deleted.
+            # pytest.param("single"),
```
Comment (tiffanyn108) [inline tests/cross_functional/scale/noobaa/test_delete_objects.py:86]: Delete_mode "single"  is not tested?

Comment (mashetty330) [reply tests/cross_functional/scale/noobaa/test_delete_objects.py:86]: We have tested and as the comment says it has heavy toll on memory consumption for the jenkins agent and also run time increases drastically. So we decided to omit this.

**Review by tiffanyn108 — APPROVED** [review]

File: tests/conftest.py [hunk 8]
```diff
@@ -7434,49 +7436,59 @@ def finalizer():
...
     Scale the noobaa pod resources and scale endpoint count
 
     """
 
-    storagecluster_obj = OCP(
-        kind="storagecluster",
-        resource_name="ocs-storagecluster",
-        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
-    )
+    def factory(min_ep_count=3, max_ep_count=3, cpu=6, memory="10Gi"):
```
Comment (keesturam) [inline tests/conftest.py:7464]: why is `min_ep_count` set to 3. why not 1.

File: ocs_ci/ocs/constants.py [hunk 9]
```diff
@@ -489,6 +489,7 @@
 CSI_CEPHFSPLUGIN_LABEL = "app=csi-cephfsplugin"
 CSI_RBDPLUGIN_LABEL = "app=csi-rbdplugin"
 OCS_OPERATOR_LABEL = "name=ocs-operator"
+ROOK_CEPH_OPERATOR_LABEL = "app=rook-ceph-operator"
```
Comment (PrasadDesala) [inline ocs_ci/ocs/constants.py:492]: we already have an existing constant, please use it - OPERATOR_LABEL

File: ocs_ci/ocs/resources/backingstore.py [hunk 10]
```diff
@@ -391,7 +391,7 @@ def _create_backingstore(method, uls_dict):
                                 cld_mgr, backingstore_name, uls_name, region
                             )
                         mcg_obj.check_backingstore_state(
-                            backingstore_name, constants.BS_OPTIMAL
+                            backingstore_name, constants.BS_OPTIMAL, timeout=timeout
```
Comment (PrasadDesala) [inline ocs_ci/ocs/resources/backingstore.py:394]: this new timeout param is missing a doc string

File: ocs_ci/ocs/resources/pod.py [hunk 11]
```diff
@@ -1526,9 +1526,11 @@ def run_io_and_verify_mount_point(pod_obj, bs="10M", count="950"):
 def get_pods_having_label(
     label,
     namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+    retry=0,
```
Comment (PrasadDesala) [inline ocs_ci/ocs/resources/pod.py:1529]: why do you need for a retry arg here?

File: ocs_ci/ocs/resources/pvc.py [hunk 12]
```diff
@@ -182,7 +182,7 @@ def resize_pvc(self, new_size, verify=False):
 
         # Modify size of PVC
         assert self.ocp.patch(
-            resource_name=self.name, params=patch_param
+            resource_name=self.name, params=patch_param, format_type="merge"
```
Comment (PrasadDesala) [inline ocs_ci/ocs/resources/pvc.py:185]: We are not seeing any issues with the resize pvc without the pass of the format_type. Do you see any issues in your case?

File: tests/conftest.py [hunk 13]
```diff
@@ -7711,3 +7723,100 @@ def finalizer():
...
+        constants.NOOBAA_DB_LABEL_47_AND_ABOVE,
+    ]
+    nb_pvc = get_all_pvc_objs(selector=constants.NOOBAA_DB_LABEL_47_AND_ABOVE)[0]
+
+    def factory(pv_size="50"):
+        """
+        Args:
+            pv_size(int): Size in GB
+
+        Returns:
```
Comment (PrasadDesala) [inline tests/conftest.py:7756]: I think you can delete the returns doc string If you are not returning anything

File: tests/cross_functional/conftest.py [hunk 14]
```diff
@@ -1070,7 +1070,13 @@ def validate_mcg_bg_features(
...
 
     def factory(
-        feature_setup_map, run_in_bg=False, skip_any_features=None, object_amount=5
+        feature_setup_map,
+        run_in_bg=False,
+        skip_any_features=None,
+        object_amount=5,
+        run_current=True,
+        retry=1,
+        delay=10,
```
Comment (PrasadDesala) [inline tests/cross_functional/conftest.py:1079]: doc strings for these two new arg is missing, please add

File: tests/cross_functional/conftest.py [hunk 15]
```diff
@@ -1207,6 +1218,8 @@ def factory(
         skip_any_type=None,
         skip_any_provider=None,
         skip_any_features=None,
+        retry=1,
+        delay=10,
```
Comment (PrasadDesala) [inline tests/cross_functional/conftest.py:1222]: doc strings for these two new arg is missing, please add

File: ocs_ci/ocs/resources/pod.py (hunk 11 above)
Comment (mashetty330) [reply ocs_ci/ocs/resources/pod.py:1529]: In my test i use the KCS article to resize noobaa db pv. during that process some of the operators are scaled down and scaled up. I use get_pods_having_label there to fetch the pod info to create pod objects, which futher used to verify pod status. This function is used immediately after the scale up step. sometimes the pods wont even be present that quicker. Having hard coded sleep might not be logical. so i extended the available retry parameter in the get() method to get_pods_having_label.

**Review by PrasadDesala — APPROVED** [review]
